specification/trace/semantic_conventions/http.md

```diff
@@ -194,6 +195,22 @@ Following attributes MUST be provided **at span creation time** (when provided a
 
 Note that in some cases host and port identifiers in the `Host` header might be different from the `server.address` and `server.port`, in this case instrumentation MAY populate `Host` header on `http.request.header.host` attribute even if it's not enabled by user.
 
+### HTTP client span duration
+
+There are some minimal constraints that SHOULD be honored:
+
+* HTTP client spans SHOULD start sometime before the first request byte is sent. This may or may not include connection time.
+* HTTP client spans SHOULD end be sometime after the HTTP response headers are fully read (or when they fail to be read). This may or may not include reading the response body.
+
+If there is any possibility for application code to not fully read the HTTP response
+(and for the HTTP client library to then have to clean up the HTTP response asynchronously),
+the HTTP client span SHOULD NOT be ended in this cleanup phase
+(and instead should end after the HTTP response headers are fully read, or fail to be read).
```

Comment:
I'd expect them to end when the call completes (when response stream is passed to user, buffered or not read yet). Any reason to recommend time-to-headers?

Reply:
That is why I would still use the more vague term "natural"...

How will libraries behave with "[Expect](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Expect): 100-continue"? It may become a mess if we overspecify...

Reply:
> I'd expect span to end when the call completes (when response stream is passed to user, either buffered or not read yet). Any reason to recommend time-to-headers?

One case I was trying to address is when an HTTP client returns a promise (for the initial response, not including the body yet) which is then never read, in which case it's not clear to me whether the "call completes" 😅.

I have made this recommendation more general, ptal.

> It may become a mess if we overspecify...

Indeed 😂

> How will libraries behave with "[Expect](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Expect): 100-continue"?

Do you think there's a problem with the current language that essentially says HTTP client spans shouldn't end before all of the response headers are received?